.github/workflows/rocm-nightly-upstream-sync.yml

```diff
@@ -0,0 +1,52 @@
+# Pulls the latest changes from upstream into main and opens a PR to merge
+# them into rocm-main branch.
+
+name: ROCm Nightly Upstream Sync
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 6 * * 1-5'
+permissions:
+  contents: write
+  pull-requests: write
+env:
+  SYNC_BRANCH_NAME: ci-upstream-sync-${{ github.run_number }}_${{ github.run_attempt }}
+jobs:
+  sync-main:
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          gh auth status
+          gh repo sync rocm/xla -b main
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  create-sync-branch:
+    needs: sync-main
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
+      - name: Create branch
+        run: |
+          git fetch
+          git checkout origin/main 
+          git checkout -b $SYNC_BRANCH_NAME
+          # Try and merge rocm-main into this new branch so that we don't run upstream's CI code
+          git config --global user.email "github-actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git merge origin/rocm-main || true
+          # If the merge creates conflicts, we want to abort and push to origin anyways so that a dev can resolve the conflicts
```

Comment:
I don't think a push in the case of failure would be what we want. AFAIK a failed merge wouldn't have any new commits because the top one would have failed to create due to conflicts.

I think a message about the failure and how to recreate (output the git commands to pull and do the merge locally) would be better

Reply:
I think we do want to push so that github will  have the `ci-upstream-sync-blahblahblah` branch that we can manually pull down and work with. I do agree that putting an error message about how to fix it with a `merge` would be good, though. Maybe as a comment on the PR that gets opened in the job below this one?

Reply:
Well I think the more important point is that if the merge fails, there won't be a new commit to push, since the commit failed to create, and won't exist until someone resolves the conflicts and manually creates the commit, which won't happen in the CI.

Your push will most likely push nothing.